### ISSUE TITLE
invalid offset in trunk

### ISSUE THREAD

**Issue by jensp** [ponchio/untrunc#34]
Hey, I have a very large (363GB) .mov file with a corrupted header. If I run untrunc against a valid file from the same app it quits with "invalid offset in trunk".

I have tried to debug the issue by printing the values from the if statement:

offsets size: 15784320
i: 1148160 
offset: -2117294592 
mdat start: 40 
mdat length: 28970628568 

With the offset being negative the comparison obviously fails. Any ideas?

**Comment by ponchio** [comment]
Ciao,
           did I read it correctly, 363GB?

You have no hope with untrunc in its current state, as it works in RAM and
has some problem I had still (sorry guys) no time to fix above 2GB, anyway.

I was planning to do a 'two pass' version, which would be able to fix
arbitrarily large files, but I have very little time  (as usual).

Basically it would make one pass to detect the offsets and another one to
write the mdat to disk, with no need to keep everything in ram.

I will do it sooner or later (hopefully sooner).

Federico

**Comment by jensp** [comment]
yep, 363 GB. The memory part isn't an issue for me, I've been using a large 1TB swapfile on a fast SSD, but yeah the 2GB limitation is probably the issue.
Anyway thanks a lot for your quick reply and for writing the tool in the first place, I'll try to find another solution in the mean time.

**Comment by jamonahn** [comment]
I "bypassed" the exit by breaking out of the for loop in track.cpp.  It worked for me on a 200 MB file.
	for(int i = 0; i < offsets.size(); i++) {
		int offset = offsets[i];
		if(offset < mdat->start || offset - mdat->start > mdat->length) {
// added break here
                        break; 
			cout << "Invalid offset in track!\n";
			exit(0);
		}

**Comment by Julusian** [comment]
I am having this issue too with a 43gb file.

What work would need to be done to allow for fixing larger files in memory or swap?
If its something simple, I would be willing to making an attempt at it
